File: 백준/Silver/11723. 집합/집합.py
# 백준 11723번 : 집합
# https://www.acmicpc.net/problem/11723
# 자료구조 : set 사용

import sys

def add(num): # S에 x를 추가
    s.add(num)

def remove(num): # S에 x 제거
    if num in s: s.remove(num)

def check(num): # S에 x가 있는지 확인
    if num in s: sys.stdout.write("1\n")
    else : sys.stdout.write("0\n")

def toggle(num): # S에 x가 있으면 제거, 없으면 추가
    if num in s: s.remove(num)
    else : s.add(num)

def all():
    global s # 할당전에 참조되는 변수이기 때문
    s.clear()
    s = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20}

def empty(): # S를 공집합으로 바꿈
    s.clear()

s = set()
m = int(input())

for i in range(m):
    string = sys.stdin.readline().rstrip()

    if string[0:3] == "add" : add(int(string[4]))
    elif string[0:6] == "remove" : remove(int(string[7]))
    elif string[0:5] == "check" : check(int(string[6]))
    elif string[0:6] == "toggle" : toggle(int(string[7]))
    elif string == "all" : all()
    elif string == "empty" : empty()

    #print(s)

File: 백준/Silver/11723. 집합/집합.java
import java.util.*;
import java.io.*;

public class Main {

	public static void add(int x) {
		set.add(x);
	}
	
	public static void remove(int x) {
		if(set.contains(x)) {
			set.remove(x);
		}
	}
	
	public static int check(int x) {
		if(set.contains(x)) {
			return 1;
		}
		else return 0;
	}
	
	public static void toggle(int x) {
		if(set.contains(x)) {
			set.remove(x);
		}
		else {
			set.add(x);
		}
	}
	
	public static void all() {
		set.clear();
		for(int i=1;i<=20;i++) {
			set.add(i);
		}
	}
	
	public static void empty() {
		set.clear();
	}
	
	static Set<Integer> set = new HashSet<Integer>();
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int tc = Integer.parseInt(br.readLine());
		int x = 0;
		
		for(int i=0;i<tc;i++){
			String[] buffer = br.readLine().split(" ");
			String word = buffer[0];
			if(buffer.length == 2) {
				x = Integer.parseInt(buffer[1]);
			}
				
			
			if(word.equals("add")) add(x);
			else if(word.equals("remove")) remove(x);
			else if(word.equals("check")) bw.write(check(x)+"\n");
			else if(word.equals("toggle")) toggle(x);
			else if(word.equals("all")) all();
			else if(word.equals("empty")) empty();
			
		}
		bw.flush();

	}

}
